File: JPA/sample-projects/e-commerce/src/main/java/me/kzv/ecommerce/repository/ItemRepositoryCustom.java
package me.kzv.ecommerce.repository;

import me.kzv.ecommerce.dto.ItemSearchDto;
import me.kzv.ecommerce.dto.MainItemDto;
import me.kzv.ecommerce.entity.Item;
import org.springframework.data.domain.Page;
import org.springframework.data.domain.Pageable;

public interface ItemRepositoryCustom {
    Page<Item> getAdminItemPage(ItemSearchDto itemSearchDto, Pageable pageable);

    Page<MainItemDto> getMainItemPage(ItemSearchDto itemSearchDto, Pageable pageable);
}

File: toy/e-commerce/src/main/kotlin/me/kzv/shop/repository/ItemRepositoryCustom.kt
package me.kzv.shop.repository

import me.kzv.shop.controller.dto.ItemSearchDto
import me.kzv.shop.controller.dto.MainItemDto
import me.kzv.shop.entity.Item
import org.springframework.data.domain.Page
import java.awt.print.Pageable

interface ItemRepositoryCustom {

    fun getAdminItemPage(itemSearchDto: ItemSearchDto, pageable: Pageable): Page<Item>

    fun getMainItemPage(itemSearchDto: ItemSearchDto, pageable: Pageable): Page<MainItemDto>
}